executor/src/witgen/symbolic_witness_evaluator.rs

```diff
@@ -51,4 +54,14 @@ where
             Ok(values[row as usize].into())
         }
     }
+
+    fn challenge<'b>(&self, challenge: &'b Challenge) -> AffineResult<&'b AlgebraicReference, T> {
+        match self.fixed_data.challenges.get(&challenge.id) {
+            Some(value) => Ok((*value).into()),
+            // The requested challenge is not yet available. Note that this means that any
```

Comment:
Shouldn't this be a panic?

Reply:
Then we'd have to filter out these identities. But yeah, that might be cleaner. Should I do that?

Reply:
I mean aren't those identities filtered out already?

Reply:
Not yet! I'll do that now.